### PR TITLE
Update fhir dependencies in backend. Update Azure Identity in function.

### LINKED ISSUE #5047: Snyk - Update High and Critical vulnerabilities (Jan 2023)

### PR DESCRIPTION
**Pull request by rin-skylight**
# DEVOPS PULL REQUEST

## Related Issue

- Resolves #5047.

## Changes Proposed

- Updates FHIR dependencies to resolve https://security.snyk.io/vuln/SNYK-JAVA-CAUHNHAPIFHIR-3245104.
- Updates function app dependencies to resolve [SNYK-JS-JSONWEBTOKEN-3180020](https://app.snyk.io/vuln/SNYK-JS-JSONWEBTOKEN-3180020).

## Additional Information

- The function app was successfully verified on `dev7`.
- The application was smoke tested on `dev7` with no adverse effects noted.

## Testing

- Dependent functionality should be verified to ensure package upgrades have not made a deleterious impact.

### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Cloud
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by mehansen — DISMISSED** [review]
lgtm! thanks Rin

File: backend/build.gradle [hunk 1]
```diff
@@ -129,6 +129,7 @@ dependencies {
     testImplementation "com.github.tomakehurst:wiremock-jre8:2.35.0"
 
     implementation 'ca.uhn.hapi.fhir:hapi-fhir-structures-r4:6.2.4'
+    implementation 'ca.uhn.hapi.fhir:org.hl7.fhir.utilities:5.6.92' //TODO: Remove when hapi-fhir-structures-r4 is upgraded to invulnerable version. See https://security.snyk.io/vuln/SNYK-JAVA-CAUHNHAPIFHIR-3245104.
```
Comment (BobanL) [inline backend/build.gradle:132]: Based on this [comment](https://github.com/hapifhir/org.hl7.fhir.core/issues/1082#issuecomment-1400975407), I'm not sure if that latest version is actually sufficient for the fix. I'm sure this one is better than the last one though!

File: ops/services/app_functions/report_stream_batched_publisher/functions/package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
   },
   "dependencies": {
     "@azure/functions": "^1.2.3",
-    "@azure/identity": "^1.5.2",
+    "@azure/identity": "^2.0.0",
```
Comment (BobanL) [inline ops/services/app_functions/report_stream_batched_publisher/functions/package.json:22]: Also I ran a depcheck against the function app and it showed that azure/identity is unused in the function app. Even going back to the original commit that added it, I don't believe it was actually ever used. I've tested the function app without it and it works!

**Review by BobanL — CHANGES_REQUESTED** [review]

**Comment by BobanL** [comment]
Holding off on upgrading `fhir.utilities` until the security vulnerability has been resolved.

**Comment by rin-skylight** [comment]
I am going to close and re-create this, as the check-graphql-types check is doing something completely bizarre. It is throwing errors that do not exist on my machine, so I will start fresh.

**Pull request closed without merging**